app/src/main/java/com/techdining/www/wallpage/view/home/HomeFragment.kt

```diff
@@ -4,29 +4,43 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.TextView
 import androidx.fragment.app.Fragment
-import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.lifecycleScope
+import com.techdining.www.wallpage.Injection
 import com.techdining.www.wallpage.R
+import com.techdining.www.wallpage.adapter.HomePageAdapter
+import com.techdining.www.wallpage.databinding.FragmentHomeBinding
 import com.techdining.www.wallpage.viewModel.HomeViewModel
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
 
 class HomeFragment : Fragment() {
 
     private lateinit var homeViewModel: HomeViewModel
+    private val adapter = HomePageAdapter()
+    private var fetchJob: Job? = null
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        homeViewModel =
-            ViewModelProvider(this).get(HomeViewModel::class.java)
-        val root = inflater.inflate(R.layout.fragment_home, container, false)
-        val textView: TextView = root.findViewById(R.id.text_home)
-        homeViewModel.text.observe(viewLifecycleOwner, Observer {
-            textView.text = it
-        })
-        return root
+        val binding = FragmentHomeBinding.inflate(inflater, container, false)
+        context ?: return binding.root
+
+        binding.photoList.adapter = adapter
+        homeViewModel = ViewModelProvider(this, Injection.provideViewModelFactory())
+            .get(HomeViewModel::class.java)
+
+        fetchJob?.cancel()
+        fetchJob = lifecycleScope.launch {
```

Comment:
do we need to cancel this task based on lifecycle?

Or should this be moved to onDestroyView?

Reply:
We need to cancel it every time we start a new job since this is being done asyncronously the previous job might be still running which will cause for the data to not be displayed properly.